src/audio/eq_fir/eq_fir.c

```diff
@@ -301,7 +369,7 @@ static int eq_fir_setup(struct comp_dev *dev, struct comp_data *cd, int nch)
 
 	/* Assign delay line to each channel EQ */
 	eq_fir_init_delay(cd->fir, cd->fir_delay, nch);
-	return 0;
+	return 1;
```

Comment:
this makes "1" something "special," easy to overlook maybe. Alternatively the caller could just check `cd->fir_delay_size` for 0, right?

Reply:
That would work yes. Maybe it's better - I'll change it.